homeassistant/components/shelly/__init__.py

```diff
@@ -184,6 +184,10 @@ def _async_input_events_handler(self):
 
     async def _async_update_data(self):
         """Fetch data."""
+
+        if self.hass.is_stopping:
```

Comment:
We shouldn't fix this here. Instead we should make sure that the data update coordinator cancels any pending update.

Reply:
Changed approach